server/fleet/integrations.go

```diff
@@ -15,120 +17,161 @@ type TeamIntegrations struct {
 	Zendesk []*TeamZendeskIntegration `json:"zendesk"`
 }
 
-// ToIntegrations converts a TeamIntegrations to an Integrations struct.
-func (ti TeamIntegrations) ToIntegrations() Integrations {
-	var intgs Integrations
-	intgs.Jira = make([]*JiraIntegration, len(ti.Jira))
-	for i, j := range ti.Jira {
-		intgs.Jira[i] = j.ToJiraIntegration()
+// MatchWithIntegrations matches the team integrations to their corresponding
+// global integrations found in globalIntgs, returning the resulting
+// integrations struct. It returns an error if any team integration does not
+// map to a global integration, but it will still return the complete list
+// of integrations that do match.
+func (ti TeamIntegrations) MatchWithIntegrations(globalIntgs Integrations) (Integrations, error) {
```

Comment:
do we need to differentiate between regular errors and errors due to integration mismatches? I wonder if returning a special error instance on which we can do `errors.Is` would be useful (specially in server/datastore/mysql/teams.go were we are just ignoring all errors)

Reply:
I don't think it's necessary at the moment, in mysql/teams we don't care about any potential errors as we're only interested in matching the team's integrations to the global ones, at this stage the global ones are necessarily valid (they can't be saved if they're not).